File: src/ServerPulse.Backend/AnalyzerApi/Domain/Dtos/Responses/LoadAmountStatisticsResponse.cs
﻿namespace AnalyzerApi.Domain.Dtos.Responses
{
    public class LoadAmountStatisticsResponse : BaseStatisticsResponse
    {
        public int AmountOfEvents { get; set; }
        public DateTime DateFrom { get; set; }
        public DateTime DateTo { get; set; }
    }
}

File: src/ServerPulse.Frontend/src/app/modules/analyzer/domain/dtos/responses/statistics/loadAmountStatisticsResponse.ts
import { BaseStatisticsResponse, LoadAmountStatistics, mapBaseStatisticsResponseToBaseStatistics } from "../../../..";

export interface LoadAmountStatisticsResponse extends BaseStatisticsResponse {
    amountOfEvents: number;
    dateFrom: Date;
    dateTo: Date;
}

export function mapLoadAmountStatisticsResponseToLoadAmountStatistics(response: LoadAmountStatisticsResponse): LoadAmountStatistics {
    return {
        ...mapBaseStatisticsResponseToBaseStatistics(response),
        amountOfEvents: response?.amountOfEvents,
        dateFrom: new Date(response?.dateFrom),
        dateTo: new Date(response?.dateTo)
    };
}